rest/admin-ui-ext/src/main/java/org/keycloak/admin/ui/rest/UIRealmResource.java

```diff
@@ -75,7 +77,7 @@ private void updateUserProfileConfiguration(UIRealmRepresentation rep) {
             return;
         }
 
-        Response response = new UserProfileResource(session, auth).update(upConfig);
+        Response response = new UserProfileResource(session, auth, adminEvent).update(upConfig);
```

Comment:
This means that always when realm is updated in the admin console `General` tab, the userProfile event would be also always sent even if there is no change in userProfile. Do we want to be a bit smart here or is the behaviour to always send event rather ok to avoid any issues with missing sent event?

In theory, we can send the userProfile event just if "migration strategy" of user profile was updated as that is the only thing of user profile updated in the `General` tab.

The opposite case (The case when only userProfile "migration strategy" is updated and none of realm settings) is likely quite error prone to try be smart and hence we likely always want to send realm update event?

I am not 100% sure for "being smart" even on userProfile event, so if you prefer current behaviour, it is fine with me.

Reply:
You are right, I didn't consider that this was called in any realm update (even not related to UP). Maybe we can implement equals in UPConfig/UPAttribute and just call the update if they are different at this point...

I would not do the same for the realm as it has been always working that way. And the same for the direct call to the UP endpoint, if someone is sending a direct update to the UP config endpoint the event is triggered even if the update is changing nothing.

WDYT?

Reply:
Sounds good! Implementing equals/hashcode in `UPConfig` and related classes in package `org.keycloak.representations.userprofile.config` looks like a good thing.

Reply:
Done! I have just added the equals/hashCode (pain Objects comparisons) methods for the UP representation classes and added a new test for the UIRealmResource endpoint. Check if you like it.